x/marketmap/keeper/keeper.go

```diff
@@ -59,22 +70,151 @@ func (k Keeper) GetAllTickers(ctx sdk.Context) ([]types.Ticker, error) {
 	return tickers, err
 }
 
+// GetAllTickersMap returns the set of Ticker objects currently stored in state
+// as a map[TickerString] -> Tickers.
+func (k *Keeper) GetAllTickersMap(ctx sdk.Context) (map[string]types.Ticker, error) {
+	iter, err := k.tickers.Iterate(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+	keyValues, err := iter.KeyValues()
+	if err != nil {
+		return nil, err
+	}
+
+	m := make(map[string]types.Ticker, len(keyValues))
+	for _, keyValue := range keyValues {
+		m[string(keyValue.Key)] = keyValue.Value
+	}
+
+	return m, nil
+}
+
 // CreateTicker initializes a new Ticker.
 // The Ticker.String corresponds to a market, and must be unique.
-func (k Keeper) CreateTicker(ctx sdk.Context, ticker types.Ticker) error {
+func (k *Keeper) CreateTicker(ctx sdk.Context, ticker types.Ticker) error {
+	// Check if Ticker already exists for the provider
+	alreadyExists, err := k.tickers.Has(ctx, types.TickerString(ticker.String()))
+	if err != nil {
+		return err
+	}
+	if alreadyExists {
+		return types.NewTickerAlreadyExistsError(types.TickerString(ticker.String()))
+	}
+	// Create the config
+	return k.tickers.Set(ctx, types.TickerString(ticker.String()), ticker)
+}
+
+// GetAllProviders returns the set of Providers objects currently stored in state.
+func (k *Keeper) GetAllProviders(ctx sdk.Context) ([]types.Providers, error) {
+	iter, err := k.providers.Iterate(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+	providers, err := iter.Values()
+	if err != nil {
+		return nil, err
+	}
+	return providers, err
+}
```

Comment:
This will return duplicates if a provider is configured to retrieve multiple tickers. should this be a distinct check instead?

Reply:
oh i see this returns the values associated with each currency pair.

Reply:
yeah this should just be removed.  The `Map` functions should only be used here.